cpp/tests/utilities/type_lists.hpp

```diff
@@ -61,8 +60,19 @@ constexpr auto types_to_ids() {
   constexpr auto N = GetSize<TYPES>;
   return types_to_ids_impl<TYPES>(std::make_index_sequence<N>());
 }
+
 }  // namespace detail
 
+template <typename TypeParam, typename T>
+auto make_type_param_vector(std::initializer_list<T> const& init_list) {
```

Comment:
Why is this necessary?
I suppose, one reason is narrowing conversions limitations. 
https://en.cppreference.com/w/cpp/language/list_initialization#Narrowing_conversions

Reply:
Yea, we need this for casting to `bool`.